docs/int_header/INT_header.md

```diff
@@ -0,0 +1,320 @@
+
+## IP Header update
+
+### Encapsulation
+
+We want to preserve the routing and other information on the packet, so we will modify the protocol (IPv4) / next header (IPv6), length and for IPv4 the header checksum.
+
+Define a new L3 protocol.  The protocol for the INT INT header will be 63 "Any local network", since the INT header should never be exposed outside of the service providers network.  There is no specification for portocol 63.
+
+IPv4:
+
+* Update the total lenght to match the new size.
+* Set the Protocol to the INT procol number. 253
+* Update the header checksum
+
+IPv6:
+
+* Set the Next Header to be the identifier for an IPv6 extension header to the INT procol number.  If exisint extension headers are in place, INT will prepend to the first extension header.
+* The INT header next header field will be the exisint IPv6 next header value.
+* The Hdr Ext Len will be updated appropriatly
+
+### Decapsulation
+
+The origional protocol / next header will be restored and the size and checksum will be recalculated as appropriate.
+
+## INT Header
+
+### INT metadata header (64 bits)
+
+This format will be used for both the device INT header and the network path INT header.  The INT instruction bitmask will indicat which data will be used.
+
+* Each INT metadata header is 8B long and should be 0 when initialized
+  * Ver (2b): INT metadata header version. Should be zero for this version
+  * Rep (2b): Replication requested. Should be zero
+  * C (1b): Copy. Should be zero
+  * E (1b): Max Hop Count exceeded. Set to 1 for the device INT header and when the max hop count has been exceeded with the networking header.
+  * Reserved (4b): Always zero.
+* Instruction Count (5b): The number of instructions that are set (i.e., number of
+1’s) in the instruction bitmap. This is 00010 (decimle 2) for the default gateway INT header.  Can be changed to add additional data.
+* Max Hop Count (8b): 1 for the device INT header and configurable for the network header.
+* Total Hop Count (8b): The current hop count for the network header and 1 for the device header.
+* INT instructions are encoded as a bitmap in the 16 bit INT Instruction field: the first 8 bits
+corresponds to a specific standard metadata as specified in Section 3 of the P4 INT spec.  The 9th-11th are defined in the metadata section below.  For the device ID the default is 000000001110000000.  If the originating device is using IPv6, then the originating IP address is not captured and the bit mask for the device INT header will be 000000001100000000.  For the network INT header the bit mask would be 100000000000000000.   Additional metadata can be added for the network INT header for performance and additional use cases.
+  * bit0 (MSB): Switch ID
+  * bit8: Origionating Device MAC (Most signifigant 4 octects)
```

Comment:
Origionating -> Originating
as with the next 2 lines
octects -> octets (elswhere in file too)